src/group_theory/group_action/defs.lean

```diff
@@ -52,17 +52,20 @@ class has_vadd (G : Type*) (P : Type*) := (vadd : G → P → P)
 @[to_additive has_vadd]
 class has_scalar (M : Type*) (α : Type*) := (smul : M → α → α)
 
+@[to_additive]
+instance has_mul.to_has_scalar (α : Type*) [has_mul α] : has_scalar α α := ⟨(*)⟩
+
 infix ` +ᵥ `:65 := has_vadd.vadd
 infixr ` • `:73 := has_scalar.smul
 
 /-- Type class for additive monoid actions. -/
-@[protect_proj] class add_action (G : Type*) (P : Type*) [add_monoid G] extends has_vadd G P :=
+@[protect_proj] class add_action (G : Type*) (P : Type*) [add_zero_class G] extends has_vadd G P :=
 (zero_vadd : ∀ p : P, (0 : G) +ᵥ p = p)
 (add_vadd : ∀ (g₁ g₂ : G) (p : P), (g₁ + g₂) +ᵥ p = g₁ +ᵥ (g₂ +ᵥ p))
 
 /-- Typeclass for multiplicative actions by monoids. This generalizes group actions. -/
 @[protect_proj, to_additive]
-class mul_action (α : Type*) (β : Type*) [monoid α] extends has_scalar α β :=
+class mul_action (α : Type*) (β : Type*) [mul_one_class α] extends has_scalar α β :=
```

Comment:
I'm a little wary of making this generalization as part of this PR - you don't need it for the lemmas here, and `mul_smul` sort of implies the associativity you're generalizing away anyway.

Can you save it for another PR?

Reply:
Absolutely! It will be much easier to get this done without it. I only included it because I eroneously thought "Eric will request this if I don't do it now 😂".

I'm curious enough to see if this will build / where it fails that I'll wait till that happens but once it does, I'll push another commit, restoring `mul_action`, but preserving the generalisation I need for `smul_mul_smul`.

Reply:
I though about doing this myself in the ring refactor, but concluded that I wasn't able to construct any instances to take advantage of it.